Index: suricata-4.0.4/configure.ac
===================================================================
--- suricata-4.0.4.orig/configure.ac
+++ suricata-4.0.4/configure.ac
@@ -75,8 +75,8 @@
     AC_PROG_LN_S
     AC_PROG_MAKE_SET
 
-    AC_PATH_PROG(HAVE_PKG_CONFIG, pkg-config, "no")
-    if test "$HAVE_PKG_CONFIG" = "no"; then
+    PKG_PROG_PKG_CONFIG
+    if test "x$PKG_CONFIG" = "x"; then
         echo
         echo "   ERROR! pkg-config not found, go get it  "
         echo "   http://pkg-config.freedesktop.org/wiki/ "
