Index: suricata/configure.ac
===================================================================
--- suricata.orig/configure.ac
+++ suricata/configure.ac
@@ -76,8 +76,8 @@
     AC_PROG_MAKE_SET
     AC_PROG_GREP
 
-    AC_PATH_PROG(HAVE_PKG_CONFIG, pkg-config, "no")
-    if test "$HAVE_PKG_CONFIG" = "no"; then
+    PKG_PROG_PKG_CONFIG
+    if test "x$PKG_CONFIG" = "x"; then
         echo
         echo "   ERROR! pkg-config not found, go get it  "
         echo "   http://pkg-config.freedesktop.org/wiki/ "
