--- a/configure.ac
+++ b/configure.ac
@@ -76,8 +76,8 @@
     AC_PROG_MAKE_SET
     AC_PROG_GREP
 
-    AC_PATH_PROG(HAVE_PKG_CONFIG, pkg-config, "no")
-    if test "$HAVE_PKG_CONFIG" = "no"; then
+    PKG_PROG_PKG_CONFIG
+    if test "x$PKG_CONFIG" = "x"; then
         echo
         echo "   ERROR! pkg-config not found, go get it  "
         echo "   http://pkg-config.freedesktop.org/wiki/ "
@@ -1465,7 +1465,7 @@
     suricata_update_rule_files="suricata-update-rule-files"
 
     if test "$enable_suricata_update" = "yes"; then
-      AC_CHECK_FILE([$srcdir/suricata-update/setup.py], [
+      AS_IF([test -e "$srcdir/suricata-update/setup.py"], [
           have_suricata_update="yes"], [])
     fi
     AM_CONDITIONAL([HAVE_SURICATA_UPDATE],
@@ -2309,7 +2309,7 @@
             else
 	          AC_SUBST([CARGO_HOME], [$CARGO_HOME])
             fi
-            AC_CHECK_FILES([$srcdir/rust/vendor], [have_rust_vendor="yes"])
+            AS_IF([test -e "$srcdir/rust/vendor"], [have_rust_vendor="yes"])
             if test "x$have_rust_vendor" = "xyes"; then
 	          rust_vendor_comment=""
             fi
